nx/include/switch/kernel/semaphore.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Kevoot
+
+#ifndef __SEMAPHORE_H
+#define __SEMAPHORE_H
+
+#include "mutex.h"
+
+#define SEM_DOWN false
+#define SEM_UP   true
+
+#define EBUSY 1
+
+typedef struct sem_t
```

Comment:
According to naming convention this should be `Semaphore`

Reply:
Do we really want to deviate that far from the posix standard names? Just more time on people writing ports if there always has to be a wrapper around everything just because of naming conventions. (Same as below)

Reply:
libnx synchronization primitives aren't POSIX based... Look at the other ones...